src/dataflow/src/source/file.rs

```diff
@@ -225,13 +225,55 @@ pub fn read_file_task<Ctor, I, Out, Err>(
     }) {
         Ok(file) => file,
         Err(err) => {
-            let _ = tx.send(Err(err));
+            if let Err(send_e) = tx.send(Err(err)) {
+                error!("Failed to send error when opening file: {:#}", send_e);
```

Comment:
This isn't an error; we should leave it as is. IIRC (though my memory is a bit hazy) there is a race where when we drop the source, the receiver on the other end of this channel might go away before this task is stopped, so some spurious messages might be sent, in which case this call will error.

That's why previously we were dropping the error on the floor.

Reply:
Okay, I will change this back then, with a comment. I was over-eager to catch all the bugs!